src/backend/app/users/user_schemas.py

```diff
@@ -136,19 +143,26 @@ class DbUserProfile(BaseUserProfile):
     user_id: int
 
     @staticmethod
-    async def update(db: Connection, user_id: int, profile_update: UserProfileIn):
-        """Update or insert a user profile."""
-
-        # Prepare data for insert or update
-        model_dump = profile_update.model_dump(
-            exclude_none=True, exclude=["password", "old_password"]
-        )
-
-        # If there are new roles, update the existing roles
-        if "role" in model_dump and model_dump["role"] is not None:
-            new_roles = model_dump["role"]
+    async def _handle_certificate_file(
+        profile: BaseModel, user_id: int
+    ) -> Optional[str]:
+        """Handle the certificate file upload and return the presigned URL."""
+        certificate_file = getattr(profile, "certificate_file", None)
+        if certificate_file:
+            s3_path = f"dtm-data/users/{user_id}/certificate/{certificate_file}"
+            try:
+                presigned_url = get_presigned_url(
```

Comment:
Have you inserted this presigned url in the DB? Pre-signed url expires after certain time.
If you have , it might be better to insert just s3_path in the db.

Reply:
@nrjadkry  I'm using a presigned URL to upload files to S3, and I've also saved the URL in the database to retrieve the file afterward.